Add reference page for default log component templates (#2689)
elastic/docs-content: solutions/observability/logs/logs-index-template-defaults.md

---
navigation_title: Default log component templates
description: Learn what the default component templates and ingest pipeline hooks for logs do.
applies_to:
  stack: all
  serverless: all
products:
  - id: observability
---

> # Default log component templates

Author: Ok, I see your point!

Elastic ships a set of reusable component templates and an optional `@custom` ingest pipeline that standardize how log data streams are mapped, indexed, and preprocessed.

> These defaults apply automatically to any data stream named `logs-*-*`, unless you override them with your own higher-priority index template.

Author: I was trying to avoid being specific with the '100' since you weren't sure and I couldn't find it anywhere 😄

> The managed `logs` index template has a priority of `100` and is automatically applied to any data stream following the `logs-*-*` index pattern, unless you override it with your own higher-priority index template.

Reviewer: LogsDB is also enabled in this index template 

<img width="816" height="489" alt="image" src="https://github.com/user-attachments/assets/abe39af0-15b2-4ca1-b678-5935fc72cf5a" />

Depending on your deployment, the `logs` index template applies one of the following data retention policies by default:

* In {{stack}} (self-managed and {{ecloud}}): lifecycle is managed by [Index Lifecycle Management (ILM)](../../../manage-data/lifecycle/index-lifecycle-management.md). By default, rollover occurs when the primary shard reaches 50 GB or the index age reaches 30 days.
* In {{serverless-full}}: lifecycle is managed by [Data Stream Lifecycle (DSL)](../../../manage-data/lifecycle/data-stream.md). By default, logs are retained for 30 days. 

Refer to [Logs index template reference](logs-index-template-reference.md) for instructions on how to view or edit the logs index template in {{kib}}.

## Component templates

The managed `logs` index template is composed of the following component templates:

### `logs@mappings`

Provides general mappings for logs data streams:

* Disables automatic date detection for string fields to avoid mis-parsing.  
* Defines ECS `data_stream.*` fields:
  * `data_stream.type`: constant_keyword, value `logs`  
  * `data_stream.dataset`: constant_keyword, for example `nginx.access` (must be ≤ 100 characters, no `-`)

>   * `data_stream.namespace`: constant_keyword, for example `production` (must be ≤ 100 characters, no `-`)

Reviewer: These can also be links to ECS, e.g. https://www.elastic.co/docs/reference/ecs/ecs-data_stream#field-data-stream-namespace

Author: Good point, added

### `logs@settings`

Configures default index settings for logs data streams: 

>   * Sets `@timestamp` to ingest time if missing.  

Reviewer: The component template references a default pipeline (`logs@default-pipeline`) which in turn sets `@timestamp` if missing.

<img width="539" height="464" alt="image" src="https://github.com/user-attachments/assets/23f04465-aff4-4535-bd6b-b1efc2d73a79" />

>   * Contains a hook to the optional [`logs@custom`](#customize-preprocessing-with-logscustom) pipeline.  

Reviewer: The `logs@default-pipeline` ingest pipeline contains a hook to the `logs@custom` ingest pipeline

>   * Sets `ignore_malformed` to `true` globally. With this setting, documents with malformed fields can be indexed without causing ingestion failures. Refer to [ignore_malformed](elasticsearch://reference/elasticsearch/mapping-reference/ignore-malformed.md) for a list of supported fields.

Reviewer: It also includes `ignore_dynamic_beyond_limit`.

<img width="494" height="478" alt="image" src="https://github.com/user-attachments/assets/2e8de5ca-a430-427c-9206-f9ba0dde0659" />

Author: Added!

### `ecs@mappings`

* Adds dynamic templates that automatically align fields with [Elastic Common Schema (ECS)](ecs://reference/index.md).  

Alignment with ECS helps ensure that dashboards, queries, and ML jobs can work consistently across different log sources.

## Customize preprocessing with `logs@custom`

Each logs data stream runs through the default ingest pipeline. However, you can use the `logs@custom` component template to customize your {{es}} indices. The `logs@custom` component template is not installed by default, but you can create a component template named `logs@custom` to override and extend default mappings or settings. Refer to [Edit the `logs` index template](../logs/logs-index-template-reference.md#custom-logs-template-edit) for more information.

## Using logs templates without naming conventions

> If you don’t follow the `logs-*-*` naming scheme, the managed logs index template will not apply automatically.

Reviewer 2: ```suggestion
If your logs data streams do not follow the `logs-*-*` naming scheme, the managed logs index template will not apply automatically.
```

> You can still use the defaults by composing standard component templates into your own index template. For example:

Reviewer 2: ```suggestion
You can still use the default component templates by adding them to your own index template. For example:
```

```json
PUT _index_template/my-logs-template
{
  "index_patterns": ["my-logs-template"],
  "composed_of": ["ecs@mappings", "logs@mappings", "logs@settings"],
  "priority": 200
}

> ```

Author: Example based on content from [this page](https://www.elastic.co/docs/manage-data/data-store/data-streams/set-up-data-stream).
